tests/CMakeLists.txt

```diff
@@ -1,37 +1,77 @@
 # Detray library, part of the ACTS project (R&D line)
 #
-# (c) 2021-2023 CERN for the benefit of the ACTS project
+# (c) 2021-2024 CERN for the benefit of the ACTS project
 #
 # Mozilla Public License Version 2.0
 
+message(STATUS "Building 'detray::test_utils' component")
+
 # Set the common C++ flags.
 include( detray-compiler-options-cpp )
+
 include_directories( SYSTEM $<TARGET_PROPERTY:covfie::core,INTERFACE_INCLUDE_DIRECTORIES> )
 include_directories( SYSTEM $<TARGET_PROPERTY:dfelibs::dfelibs,INTERFACE_INCLUDE_DIRECTORIES> )
 
-# Set up a test library, which the "new style" benchmarks and tests could use.
-file( GLOB _detray_test_headers
+# Set up a test library, which can be used by downstream projects.
+file( GLOB _detray_test_utils_headers
    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-   "include/detray/test/common/*.hpp"
-   "include/detray/test/common/detail/*.hpp"
-   "include/detray/test/common/utils/*.hpp")
+   "include/detray/test/utils/*.hpp"
+   "include/detray/test/utils/detectors/*.hpp"
+   "include/detray/test/utils/simulation/event_generator/*.hpp"
+   "include/detray/test/utils/simulation/*.hpp" )
 
-add_library( detray_test_common INTERFACE "${_detray_test_headers}" )
-add_library( detray::test_common ALIAS detray_test_common )
+add_library( detray_test_utils INTERFACE "${_detray_test_headers}" )
+add_library( detray::test_utils ALIAS detray_test_utils )
 
-target_include_directories( detray_test_common
+target_include_directories( detray_test_utils
    INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include" )
 
-target_link_libraries( detray_test_common INTERFACE GTest::gtest vecmem::core detray::core detray::csv_io detray::svgtools )
+target_link_libraries( detray_test_utils INTERFACE vecmem::core detray::core detray::detectors detray::csv_io detray::svgtools )
+
+unset( _detray_test_utils_headers )
+
+# Build the common (host and device) test library (mostly for integration tests)
+if( DETRAY_BUILD_TESTING )
```

Comment:
Is this file ever loaded if this is false?

Reply:
I hope so. This should be the case when only the test utils are required, but no actual tests are built